src/validation/utils.ts

```diff
@@ -173,6 +173,55 @@ function parsePort(data: any): Port {
   return data;
 }
 
+/**
+ * Seed nodes expected to be of form 'nodeId1@host:port;nodeId2@host:port;...'
+ * By default, any specified seed nodes (in CLI option, or environment variable)
+ * will overwrite the default nodes in src/config.ts.
+ * Special flag `<defaults>` indicates that the default seed
+ * nodes should be added to the starting seed nodes instead of being overwritten.
+ */
+function parseSeedNodes(rawSeedNodes: string): [NodeMapping, boolean] {
+  const seedNodeMappings: NodeMapping = {};
+  let defaults = false;
+  // If specifically set no seed nodes, then ensure we start with none
+  if (rawSeedNodes === '') return [seedNodeMappings, defaults];
+  const semicolonSeedNodes = rawSeedNodes.split(';');
+  for (const rawSeedNode of semicolonSeedNodes) {
+    // Empty string will occur if there's an extraneous ';' (e.g. at end of env)
+    if (rawSeedNode === '') continue;
+    // Append the default seed nodes if we encounter the special flag
+    if (rawSeedNode === '<defaults>') {
+      defaults = true;
+      continue;
+    }
+    let idHostPort: URL;
+    try {
+      idHostPort = new URL(`pk://${rawSeedNode}`);
+    } catch (e) {
+      if (e instanceof TypeError) {
+        throw new validationErrors.ErrorParse(
+          `Seed node ${rawSeedNode} must be of format 'nodeId@host:port'`,
+        );
+      }
+      throw e;
+    }
+    console.log(idHostPort);
+    const rawNodeId = idHostPort.username;
+    const rawHost = idHostPort.hostname.replace(/\[|\]/g, ''); // strip square brackets
+    const rawPort = idHostPort.port;
+    if (rawNodeId == '' || rawHost == '' || rawPort == '') {
+      throw new validationErrors.ErrorParse(
+        `Seed node ${rawSeedNode} must be of format 'nodeId@host:port'`,
+      );
+    }
+    seedNodeMappings[parseNodeId(rawNodeId)] = {
+      host: parseHostOrHostname(rawHost),
+      port: parsePort(rawPort),
+    };
+  }
+  return [seedNodeMappings, defaults];
+}
+
```

Comment:
In #310 I noticed `NodeMapping` type is not quite correct. And I changed it to `Record<NodeIdEncoded, NodeAddress`. 

Since you've moved this here, I'm going to just copy this part into `validation/utils.ts` as well, and so when rebasing this, it will just need to be replaced by what already exists in #310 when it merges.

Reply:
I've moved it into #310.